### DISCUSSION TITLE
setImage does not display animated GIFs

### DISCUSSION THREAD

**Discussion by pre-martin** [elgatosf/elgato#128] [Help] [answered]
Hi all,

the Stream Deck supports animated GIF files. On any action, you can use "Set [Icon] from File", select an animated GIF and the device will display the animated GIF.

The problem is with the plugin API: Loading the same animated GIF in a plugin, converting it to base64 and sending it via the event "setImage" to the Stream Deck, it only displays the very first frame of the animated GIF, but not the animation itself.

I stripped down the code in my plugin to this, to eliminate any errors in my plugin code:

```
var bytes = File.ReadAllBytes(@"images\red-orange.gif");
var encoded = "data:image/gif;base64," + Convert.ToBase64String(bytes);
await setImageAsync(encoded);
```

Is there any way to display animated GIFs from a plugin?

**Comment by ChekTek** [comment, marked as answer]
Hi @pre-martin, 

Unfortunately, we cannot use animated images via the API at this time. If you could provide some information for your specific use case, I would love to add your post to our internal ticket. 

For now, you can call `setImage` multiple times to simulate animation, however I would recommend using this sparingly, as it is quite inefficient.

Thanks!

**Comment by pre-martin** [comment]
Hi,

thank you for your answer.

I have a plugin that makes data from driving simulators (racing games likes Assetto Corsa Competizione, iRacing, F1 24, but also truck simulators like American Truck Simulator, or Farming Simulator) available on the Stream Deck. In the same time, the simulators can be controlled via the Stream Deck keys. For specific use cases (e.g. turn indicators, racing flags), animated images would be awesome. However, I cannot use the built-in function ("Set Icon From File"), because I also have use cases, where I have to display values on top of these images. Just "setTitle" on top of the images is not enough. So I have to load the images in my plugin and send them via `setImage` to the Stream Deck.

The animations are not with too high frequency (turn indicator could be once per second, flags would be of course smoother with more fps). So to summarize, I would have to implement some "ticker" in my plugin to switch the images by myself.

I know that changing the images in too high frequency is not efficient, because I already debugged my USB bus :-)

Regards,
Martin

**Reply by ChekTek** [reply]
Awesome! Thanks, I will share this use case with the team.
